src/test/ui/type/type-ascription-instead-of-initializer.stderr

```diff
@@ -11,15 +11,12 @@ error[E0061]: this function takes 1 argument but 2 arguments were supplied
   --> $DIR/type-ascription-instead-of-initializer.rs:2:12
    |
 LL |     let x: Vec::with_capacity(10, 20);
-   |            ^^^^^^^^^^^^^^^^^^ --  -- supplied 2 arguments
-   |            |
-   |            expected 1 argument
+   |            ^^^^^^^^^^^^^^^^^^     -- argument unexpected
    |
-note: associated function defined here
-  --> $SRC_DIR/alloc/src/vec/mod.rs:LL:COL
+help: remove the extra argument
    |
-LL |     pub fn with_capacity(capacity: usize) -> Self {
-   |            ^^^^^^^^^^^^^
+LL |     let x: Vec::with_capacity(10);
+   |            ~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
We are losing the span pointing at the associated fn here. Note to self: check where that regression might be happening.

Reply:
That would be because somehow we changed from `def_ident_span` to `get_if_local`. Fixing...